sphinx/cmd/build.py

```diff
@@ -94,190 +97,216 @@ def jobs_argument(value: str) -> int:
+            error = warning
+
+        if args.nitpicky:
+            args.confoverrides['nitpicky'] = True
+
+        app = None
+        try:
+            confdir = args.confdir or args.sourcedir
+            with patch_docutils(confdir), docutils_namespace():
+                app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
+                             args.doctreedir, args.builder, args.confoverrides,
+                             status, warning, args.freshenv, args.warningiserror,
+                             args.tags, args.verbosity, args.jobs, args.keep_going)
+                app.build(args.force_all, args.filenames)
+                return app.statuscode
+        except (Exception, KeyboardInterrupt) as exc:
+            handle_exception(app, args, exc, error)
+            return 2
 
-    if args.force_all and filenames:
-        parser.error(__('cannot combine -a option and filenames'))
+    def add_parser(self, parser):
+        return self._add_parser(parser)
 
-    if args.color == 'no' or (args.color == 'auto' and not color_terminal()):
-        nocolor()
+    def run(self, args):
+        return self._run(args)
 
-    status = sys.stdout
-    warning = sys.stderr
-    error = sys.stderr
 
-    if args.quiet:
-        status = None
+def build_main(argv: List[str] = sys.argv[1:]) -> int:
+    """Sphinx build "main" command-line entry."""
+    desc = re.sub(r'(sphinx) (\w+)', r'\1-\2', BuildCommand.__doc__)
+    parser = argparse.ArgumentParser(
+        usage='usage: %(prog)s [OPTIONS] SOURCEDIR OUTPUTDIR [FILENAMES...]',
+        epilog='For more information, visit <http://sphinx-doc.org/>.',
+        description=desc)
 
-    if args.really_quiet:
-        status = warning = None
+    parser.add_argument('--version', action='version', dest='show_version',
+                        version='%%(prog)s %s' % __display_version__)
+    parser = BuildCommand._add_parser(parser)
 
-    if warning and args.warnfile:
-        try:
-            warnfp = open(args.warnfile, 'w')
-        except Exception as exc:
-            parser.error(__('cannot open warning file %r: %s') % (
-                args.warnfile, exc))
-        warning = Tee(warning, warnfp)  # type: ignore
-        error = warning
-
-    confoverrides = {}
-    for val in args.define:
-        try:
-            key, val = val.split('=', 1)
-        except ValueError:
-            parser.error(__('-D option argument must be in the form name=value'))
-        confoverrides[key] = val
+    try:
+        args = parser.parse_args(argv)
+    except SystemExit as err:
+        return err.code
 
-    for val in args.htmldefine:
-        try:
-            key, val = val.split('=')
-        except ValueError:
-            parser.error(__('-A option argument must be in the form name=value'))
-        try:
-            val = int(val)
-        except ValueError:
-            pass
-        confoverrides['html_context.%s' % key] = val
+    return BuildCommand._run(args)
```

Comment:
We need to discuss `sphinx build` (new stye) is same as `sphinx-build` (old style). I still don't have an answer.

Reply:
The only behavior change I've suggested so far is to go from:

    sphinx-build sourcedir builddir [file, ...]

to

    sphinx build sourcedir [file, ...] [-o builddir]

But perhaps there are other options that we think are badly named and should be renamed now? I hadn't thought of that so good idea :+1:

Reply:
It's just an idea, `sphinx build html` is so so good to me.